specification/dartLangSpec.tex

```diff
@@ -19702,52 +19718,175 @@ \subsection{Type Aliases}
+\LMHash{}%
+\BlindDefineSymbol{D, F, T}%
+Let $D$ be a type alias declaration of the form
+
+\noindent
+\code{\TYPEDEF{} $F$<\TypeParametersStd> = $T$;}
+
+\noindent
+and let \DefineSymbol{U} be a parameterized type of the form
+\code{$F$<\List{U}{1}{s}>}
+in a scope where $F$ denotes $D$.
+Assume that the transitive alias expansion of $U$ is
+a type of the form $C$ or \code{$p$.$C$}
+where $C$ is an identifier denoting a class or mixin,
+and $p$ is an identifier denoting an import prefix,
+optionally followed by \synt{typeArguments}.
+Assume that $U$ occurs in an expression $e$ of the form
+`\code{$U$.\id\;\metavar{args}}'
+where \metavar{args} is derived from \syntax{<argumentPart>?},
+such that \id{} is the name of
+a static member of $C$ respectively \code{$p$.$C$}.
+The expression $e$ is then treated as
+`\code{$C$.\id\;\metavar{args}}'
+respectively
+`\code{$p$.$C$.\id\;\metavar{args}}'.
+
+\commentary{%
+This means that it is possible to use a type alias
+to invoke a static member of a class or a mixin.
+For example:%
 }
 
+\begin{dartCode}
+\CLASS{} C<X> \{
+  static void staticMethod() \{\}
+\}
+\\
+\TYPEDEF{} F = C<int>;
+\\
+\VOID{} main() \{
+  F.staticMethod(); // \comment{OK.}
+\}
+\end{dartCode}
+
 \rationale{%
-This indirectly allows an invocation of a static method to pass
-a list of actual type arguments to the class.
-This is currently an error when it occurs directly
-(e.g., \code{List<int>.castFrom(xs)}).
-But it may be part of a future language extension to allow
-static methods to use the type parameters declared by the enclosing class,
-in which case both the direct and indirect approach will be allowed.
+Note that the type arguments passed to $C$ respectively \code{$p$.$C$}
+are erased,
+such that the resulting expression can be a correct static member invocation.
+If a future version of Dart allows type arguments to be passed via the class
+in an invocation of a static member,
+it may be preferable to preserve these type arguments.
 At that time,
-existing invocations where type arguments are passed indirectly will not break,
-because it is currently an error for a static method to depend on the value
+existing invocations where type arguments are passed in this manner
+will not break,
+because it is currently an error for a static member to depend on the value
 of a formal type parameter of the enclosing class.%
 }
 
-%% TODO(eernst): Move specification of generic type aliases to this
-%% section, change the non-generic case to a special case.
-\commentary{%
-The generic variants of type alias declarations are specified
-in the section about generics
-(\ref{generics}).%
-}
+\LMHash{}%
+\BlindDefineSymbol{D, F, X_j, B_j}%
+Let $D$ be a type alias declaration of the form
+
+\noindent
+\code{\TYPEDEF{} $F$<\TypeParametersStd> = $T$;}
+
+\noindent
+\BlindDefineSymbol{Y_j}%
+Let \List{Y}{1}{s} be fresh type variables,
+assumed to satisfy the bounds of $D$.
+We say that $D$
```

Comment:
Why not just say "... if the transitive alias expansion of `T` is `X_j` for some `$j` in `1.. s`?

Reply:
That would probably be because I complained about that.
If we describe a generic type alias as "a function from types to a type", then I'm very concerned about doing things to the body of a function without actually applying the function.
If we just look at it as terms and variable-replacements, then defining the alias expansion on the RHS by itself (with unbound type variables) makes sense. 
We used the "function" description, so I wanted it to be consistent.

Reply:
Ok.  I don't really understand how this makes a difference though - either way you're evaluating an open term, either we understand how to do that or we don't.  Making it open WRT fresh type variables instead of the ones bound in the definition seems completely isomorphic to me.

Reply:
> That would probably be because I complained about that.

Exactly. ;-)

In this particular case you could also say that it makes sense to specify what it means for a type alias named `F` to expand to a type variable in terms of `F`.

I did not make any changes to this location in https://github.com/dart-lang/language/pull/1555 after all, but I did add `$s >0$` at this point, because the case where `$s$` is zero cannot occur anyway.